Dockerfiles/windows-servercore21H2.dockerfile

```diff
@@ -0,0 +1,53 @@
+# escape=`
+FROM mcr.microsoft.com/dotnet/framework/sdk:4.8-windowsservercore-ltsc2022
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'Continue'; $verbosePreference='Continue';"]
+
+# Install chocolatey
+RUN Set-ExecutionPolicy Bypass -Scope Process -Force; `
+    [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; `
+    iex ((New-Object System.Net.WebClient).DownloadString('https://community.chocolatey.org/install.ps1'))
+
+# Install git and python3.11
+RUN choco install -y git.install gnupg
+RUN choco install -y python --version=3.11.0
+
+# Install spack requirements
+RUN python -m pip install --upgrade pip setuptools wheel
+RUN python -m pip install pyreadline boto3 pyyaml pytz minio requests clingo pywin32
+
+# Restore the default Windows shell for correct batch processing.
+SHELL ["cmd", "/S", "/C"]
+
+# Install build tools including MSVC, CMake, Win-SDK
+RUN `
+    curl -SL --output vs_buildtools.exe https://aka.ms/vs/17/release/vs_buildtools.exe `
+    && start /w vs_buildtools.exe --quiet --wait --norestart --nocache `
+    --installPath "C:\Spack\BuildTools" `
```

Comment:
Change to default installation location

"%ProgramFiles(x86)%\Microsoft Visual Studio\2022\BuildTools"